pulp_2to3_migration/app/pre_migration.py

```diff
@@ -355,8 +362,47 @@ def pre_migrate_all_without_content(plan):
             # filter by repo type
             repos_to_check = plan.type_to_repo_ids[plugin_plan.type]
 
-            mongo_repo_q = mongo_Q(repo_id__in=repos_to_check)
-            mongo_repo_qs = Repository.objects(mongo_repo_q)
+            epoch = datetime.utcfromtimestamp(0)
+            # figure out which repositories/importers/distributors were updated since the last run
+            repo_premigrated_last_by_added = Pulp2Repository.objects.aggregate(
+                Max('pulp2_last_unit_added')
+            )['pulp2_last_unit_added__max'] or epoch
+            repo_premigrated_last_by_removed = Pulp2Repository.objects.aggregate(
+                Max('pulp2_last_unit_removed')
+            )['pulp2_last_unit_removed__max'] or epoch
+            imp_premigrated_last = Pulp2Importer.objects.aggregate(
+                Max('pulp2_last_updated')
+            )['pulp2_last_updated__max'] or epoch
+            dist_premigrated_last = Pulp2Distributor.objects.aggregate(
+                Max('pulp2_last_updated')
+            )['pulp2_last_updated__max'] or epoch
+
+            is_content_added_q = mongo_Q(last_unit_added__gte=repo_premigrated_last_by_added)
+            is_content_removed_q = mongo_Q(last_unit_removed__gte=repo_premigrated_last_by_removed)
+            is_new_enough_repo_q = is_content_added_q | is_content_removed_q
+            is_empty_repo_q = (
+                mongo_Q(last_unit_added__exists=False) & mongo_Q(last_unit_removed__exists=False)
```

Comment:
It might be enough to check last_unit_added__exists=False only. I guess no units can be removed if they hadn't been added to a repo in the first place.

Reply:
let's do that. other repos that used to contain content and then it got removed would be handled on the upper lines.